pkg/controller/networkpolicy/clustergroup.go

```diff
@@ -176,27 +173,62 @@ func (n *NetworkPolicyController) syncInternalGroup(key string) error {
 		klog.V(2).Infof("Internal group %s not found.", key)
 		return nil
 	}
-
 	grp := grpObj.(*antreatypes.Group)
-	// Find all Pods matching its selectors and update store.
-	groupSelector := grp.Selector
-	pods, _ := n.processSelector(groupSelector)
-	memberSet := controlplane.GroupMemberSet{}
-	for _, pod := range pods {
-		if pod.Status.PodIP == "" {
-			// No need to insert Pod IPAddress when it is unset.
-			continue
+	if grp.IPBlock == nil {
+		// Find all Pods matching its selectors and update store.
+		groupSelector := grp.Selector
+		pods, _ := n.processSelector(groupSelector)
+		memberSet := controlplane.GroupMemberSet{}
+		for _, pod := range pods {
+			if pod.Status.PodIP == "" {
+				// No need to insert Pod IPAddress when it is unset.
+				continue
+			}
+			memberSet.Insert(podToGroupMember(pod, true))
 		}
-		memberSet.Insert(podToGroupMember(pod, true))
+		// Update the internal Group object in the store with the Pods as GroupMembers.
+		updatedGrp := &antreatypes.Group{
+			UID:          grp.UID,
+			Name:         grp.Name,
+			Selector:     grp.Selector,
+			GroupMembers: memberSet,
+		}
+		klog.V(2).Infof("Updating existing internal Group %s with %d GroupMembers", key, len(memberSet))
+		n.internalGroupStore.Update(updatedGrp)
+	}
+	// Retrieve the ClusterGroup corresponding to this key.
+	cg, err := n.cgLister.Get(grp.Name)
```

Comment:
Why is this necessary? we could just filter using `updatedGrp.Name`?

Reply:
this is not needed in this PR.. but when you get a chance to review the CG Status PR, you'll see that we need the CG object to update the status once members are realized. I just wanted to have min conflicts ;)

Reply:
Okay thanks for the explanation